File: apps/pos/src/lib/order-type-label.ts
/**
 * Human label for order type in list/detail.
 * - dine_in → "Dine-in"
 * - takeaway, no partner → "Takeaway"
 * - takeaway + partner, no ref → "{title}"
 * - takeaway + partner + ref → "{title} / {ref}"
 *
 * Only show " / {ref}" when deliveryExternalRef is a non-empty string after trim.
 * Prefer deliveryPartnerTitle; if title is null/empty but partner id is set, fall back is NOT required — treat as no partner (show "Takeaway").
 * Ignore partner fields when type is dine_in.
 */
export function formatOrderTypeLabel(order: {
  type: string;
  deliveryPartnerTitle?: string | null;
  deliveryExternalRef?: string | null;
}): string {
  if (order.type !== 'takeaway') {
    return 'Dine-in';
  }
  const title = order.deliveryPartnerTitle?.trim();
  if (!title) {
    return 'Takeaway';
  }
  const ref = order.deliveryExternalRef?.trim();
  return ref ? `${title} / ${ref}` : title;
}

File: apps/android/app/src/main/java/com/spicyhome/pos/util/OrderTypeLabel.kt
package com.spicyhome.pos.util

/**
 * Human label for order type in list/detail — the Android Kotlin twin of
 * `apps/pos/src/lib/order-type-label.ts` (canonical TypeScript implementation).
 *
 * **Keep the semantics in sync with the TS file** — mirror the cases in
 * `apps/pos/src/lib/order-type-label.test.ts` and
 * `apps/android/app/src/test/java/com/spicyhome/pos/util/OrderTypeLabelTest.kt`.
 *
 * - dine_in → "Dine-in"
 * - takeaway, no partner → "Takeaway"
 * - takeaway + partner, no ref → "{title}"
 * - takeaway + partner + ref → "{title} / {ref}"
 *
 * Only show " / {ref}" when the external ref is a non-empty string after trim.
 * Prefer the partner title; if the title is null/blank, treat as no partner
 * (show "Takeaway") — no fallback to the partner id. Ignore partner fields
 * when type is dine_in.
 */
object OrderTypeLabel {

    fun format(
        type: String,
        deliveryPartnerTitle: String? = null,
        deliveryExternalRef: String? = null,
    ): String {
        if (type != "takeaway") {
            return "Dine-in"
        }
        val title = deliveryPartnerTitle?.trim()
        if (title.isNullOrEmpty()) {
            return "Takeaway"
        }
        val ref = deliveryExternalRef?.trim()
        return if (ref.isNullOrEmpty()) title else "$title / $ref"
    }
}
